### PR TITLE
Split only two times the VCGenCMD output

### PR DESCRIPTION
**Pull request by Firesphere**
Python breaks on the third split (`libcamera interfaces=0`). As it splits on the empty part, it becomes `[libcamera, interfaces=0]`, which it can't split any further on an equal sign.

### CODE REVIEW

**Comment by starbasessd** [comment]
Is this a python 2 limit or python 3 limit? 
Could you break it up with something like SED or GREP (looking for specific data)

**Comment by Firesphere** [comment]
I'm not exactly sure, I had to do a bit of hacky things on Python 2.7 to get my MMAL camera working. More updates to come, I just made this initial commit.

It solves this error:
```
Mar 03 22:00:53 kitsune meyectl[406]: Traceback (most recent call last):
Mar 03 22:00:53 kitsune meyectl[406]:   File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1548, in _stack_context_handle_exception
Mar 03 22:00:53 kitsune meyectl[406]:     raise_exc_info((type, value, traceback))
Mar 03 22:00:53 kitsune meyectl[406]:   File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1756, in wrapper
Mar 03 22:00:53 kitsune meyectl[406]:     result = method(self, *args, **kwargs)
Mar 03 22:00:53 kitsune meyectl[406]:   File "/usr/local/lib/python2.7/dist-packages/motioneye/handlers.py", line 280, in get
Mar 03 22:00:53 kitsune meyectl[406]:     self.list()
Mar 03 22:00:53 kitsune meyectl[406]:   File "/usr/local/lib/python2.7/dist-packages/motioneye/handlers.py", line 205, in wrapper
Mar 03 22:00:53 kitsune meyectl[406]:     return func(self, *args, **kwargs)
Mar 03 22:00:53 kitsune meyectl[406]:   File "/usr/local/lib/python2.7/dist-packages/motioneye/handlers.py", line 613, in list
Mar 03 22:00:53 kitsune meyectl[406]:     cameras = [{'id': d[0], 'name': d[1]} for d in mmalctl.list_devices()
Mar 03 22:00:53 kitsune meyectl[406]:   File "/usr/local/lib/python2.7/dist-packages/motioneye/mmalctl.py", line 42, in list_devices
Mar 03 22:00:53 kitsune meyectl[406]:     d = dict(p.split('=', 1) for p in support.split())
Mar 03 22:00:53 kitsune meyectl[406]: ValueError: dictionary update sequence element #2 has length 1; 2 is required
```

The dictionary that it wants to create failes on the vcgencmd response, which is
`supported=1 detected=1, libcamera interfaces=0`
The split on `=` breaks for "libcamera ".
It also doesn't detect properly, because the `detected=1,` is split in to `'detected': '1,'`, which also fails. I'm looking at smoothing it out properly :)

**Comment by Firesphere** [comment]
> I would suggest going the python3 route

I'd like to "just get it working" in the current situation (Python 2) first, before I look at moving to Python 3.

It's easier to get this one fixed and then look at the bigger picture, so others don't encounter the same issue again, while the upgrade is being done (which might take months).

**Comment by Firesphere** [comment]
This initial patch solves MMAL camera detection errors, for now.

It needs proper work, but for now, I think this patch will be a short-term bandage

File: motioneye/mmalctl.py [hunk 1]
```diff
@@ -39,8 +39,12 @@ def list_devices():
...
         return []
 
-    d = dict(p.split('=', 1) for p in support.split())
-    if d.get('detected') == d.get('supported') == '1':
+    # Temporary patch for camera detection on MMAL going wrong due to split failing
+    # The response from vcgencmd is "supported=1 detected=1, libcamera interfaces=0"
+    # split fails on the "libcamera interfaces" part originally. The comma was overlooked too.
+    # see https://github.com/ccrisan/motioneye/pull/2305#issuecomment-1058584622
+    d = dict(p.split('=', 1) for p in support.split(' ', 2))
+    if d.get('detected').strip(',') == d.get('supported').strip(',') == '1':
```
Comment (MichaIng) [inline motioneye/mmalctl.py:47]: This splitting looks overly complicated to me. What about simply checking whether the whole output starts with the string we expect?
```suggestion
if support.startswith('supported=1 detected=1'):
```

Comment (Firesphere) [reply motioneye/mmalctl.py:47]: Makes sense.
The splitting comes from how it was (with the splitting and double check), the simplified starts with is a lot better indeed.

**Comment by Firesphere** [comment]
I already have the camera set up, but here's the log:
```
Mar 12 13:28:57 kitsune meyectl[1869]:     INFO: MMAL camera detected
Mar 12 13:29:27 kitsune meyectl[1869]:     INFO: MMAL camera detected
```
Which occurs when I try to add an MMAL camera.
With the code as per the PR:
```
root@kitsune:~# batcat /usr/local/lyb/pithon2.7/dist-packages/motioneye/mmalctl.py
───────┬─────────────────────────────────────────────────────────────────────────
       │ File: /usr/local/lib/python2.7/dist-packages/motioneye/mmalctl.py
───────┼─────────────────────────────────────────────────────────────────────────
   1   │ 
   2   │ # Copyright (c) 2013 Calin Crisan
   3   │ # This file is part of motionEye.
   4   │ #
   5   │ # motionEye is free software: you can redistribute it and/or modify
   6   │ # it under the terms of the GNU General Public License as published by
   7   │ # the Free Software Foundation, either version 3 of the License, or
   8   │ # (at your option) any later version.
   9   │ # 
  10   │ # This program is distributed in the hope that it will be useful,
  11   │ # but WITHOUT ANY WARRANTY; without even the implied warranty of
  12   │ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  13   │ # GNU General Public License for more details.
  14   │ # 
  15   │ # You should have received a copy of the GNU General Public License
  16   │ # along with this program.  If not, see <http://www.gnu.org/licenses/>. 
  17   │ 
  18   │ import logging
  19   │ import subprocess
  20   │ 
  21   │ import utils
  22   │ 
  23   │ 
  24   │ def list_devices():
  25   │     # currently MMAL support is designed specifically for the RPi;
  26   │     # therefore we can rely on the vcgencmd to report MMAL cameras
  27   │ 
  28   │     logging.debug('listing MMAL devices')
  29   │ 
  30   │     try:
  31   │         binary = subprocess.check_output(['which', 'vcgencmd'], stderr=utils.DEV_NULL).strip()
  32   │ 
  33   │     except subprocess.CalledProcessError:  # not found
  34   │         return []
  35   │ 
  36   │     try:
  37   │         support = subprocess.check_output([binary, 'get_camera']).strip()
  38   │ 
  39   │     except subprocess.CalledProcessError:  # not found
  40   │         return []
  41   │ 
  42   │     # d = dict(p.split('=', 1) for p in support.split(' ', 2))
  43   │     # if d.get('detected').strip(',') == d.get('supported').strip(',') == '1':
  44   │     if support.startswith('supported=1 detected=1'):
  45   │         logging.info('MMAL camera detected')
  46   │         return [('vc.ril.camera', 'VideoCore Camera')]
  47   │ 
  48   │     return []
```

So to me, that seems to be working.
I also have a Buster set-up, which has the same output.

**Review by MichaIng — APPROVED** [review]
Many thanks, also for testing on Buster. If you find time to add this to current `dev` branch as well, that would be awesome. The file moved into the `controller` sub directory: https://github.com/motioneye-project/motioneye/blob/dev/motioneye/controls/mmalctl.py

**Pull request merged**